Tests/SwiftWarningControlTest/WarningControlTests.swift

```diff
@@ -236,6 +287,7 @@ private func assertWarningGroupControl(
   _ markedSource: String,
   diagnosticGroupID: DiagnosticGroupIdentifier,
   states: [String: WarningGroupBehavior?],
+  enclosingControls: [DiagnosticGroupIdentifier: WarningGroupBehavior] = [:],
```

Comment:
Would it make sense to move `enclosingControls` after `markedSource`. That way you start by specifying the test input followed by the expected states instead of interleaving them.